cfe_internal/core/watchdog/watchdog.cf

```diff
@@ -23,21 +26,111 @@ bundle agent cfe_internal_core_watchdog(state)
 
+    use_cfe_internal_core_watchdog_windows::
+      "any" usebundle => cfe_internal_core_watchdog_windows( $(state) );
+
   reports:
     DEBUG|DEBUG_cfe_internal_core_watchdog::
       "DEBUG $(this.bundle): Watchdog '$(state)'";
       "DEBUG $(this.bundle): Invalid state '$(state)' only enabled|disabled allowed"
         ifvarclass => "invalid_state";
 
-    !(use_cfe_internal_core_watchdog_cron_d|aix)::
-      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle and AIX hosts.";
+    !(use_cfe_internal_core_watchdog_cron_d|use_cfe_internal_core_watchdog_aix|use_cfe_internal_core_watchdog_windows)::
+      "WARNING $(this.bundle): Currently only supports /etc/cron.d on systems that have pgrep in the the stdlib paths bundle, AIX and Windows hosts.";
+}
+
+bundle agent cfe_internal_core_watchdog_windows(state)
+# @brief Manage watchdog state on windows
+# @param state enabled|disabled
+# - When enabled a scheduled task "CFEngine-watchdog" will be present and enabled
+# - When disabled a scheduled task named "CFEngine-watchdog" will be absent.
+{
+
+  vars:
+    windows::
+      "_requested_state" string => ifelse( regcmp( "enabled|disabled", $(state) ), "$(state)", "invalid");
+      "_taskname" string => "CFEngine-watchdog";
+	    "_taskfreq" string => "1";
+      "_taskscript" string => "$(sys.bindir)$(const.dirsep)watchdog.ps1";
+      "_taskrun" string => "PowerShell";
+      "_taskrun_args" string => "-NoProfile -ExecutionPolicy bypass -File";
+      "_logfile" string => "$(cfe_internal_core_watchdog._logfile)";
+      # -NonInteractive?
+
+      "_cmd_task_schedule"
+        string => `$(sys.winsysdir)$(const.dirsep)schtasks.exe /create /tn "$(_taskname)" /tr "$(_taskrun) $(_taskrun_args) '$(_taskscript)'" /ru "System" /sc minute /mo $(_taskfreq) /rl highest /f`;
+
+      # We use XML output because it's the most portable output considering localization etc .. considering localization etc ...
+      "_cmd_task_query"
+        string => `schtasks /QUERY /TN "$(_taskname)" /XML 2> $(const.dollar)null`;
+
+      "_cmd_task_query_result"
+        string => execresult( $(_cmd_task_query), powershell);
+
+      # This regular expression is used to match against the XML output querying the task
+      # We escape _taskscript with \Q \E because:
+      #    error: Regular expression error: pcre_compile() 'unknown property name after \P or \p' in expression '.*Task To Run:.*powershell -noprofile -executionpolicy bypass -file C:\Program\ Files\Cfengine\watchdog.ps1.*Scheduled Task State:.*Enabled.*Schedule Type:.*OneTimeOnly, Minute.*Repeat: Every:.*1 Minutes.*' (offset: 72)
+      "_scheduled_task_regex"
+        string => concat(".*Interval.PT$(_taskfreq)M..Interval",
+							           ".*Command.$(_taskrun)..Command",
+							           ".*Arguments.$(_taskrun_args) .\Q$(_taskscript)\E...Arguments", # this line changed
```

Comment:
TABs in all these lines, also stale comment in this line

Reply:
Ack.